File: sort/bubblesort/sort.py
#!/usr/bin/env python

def sort(arr : list[int]) -> list[int]:
    n = len(arr)
    for i in reversed(range(n)):
        swapped = False
        for j in range(i):
            if arr[j] > arr[j+1]:
                swapped = True
                arr[j],arr[j+1] = arr[j+1], arr[j]

        if not swapped:
            return arr
        
    return arr

if __name__ == "__main__":
    list = [int(x) for x in input().split()]
    print(" ".join(map(str, sort(list))))

File: sort/bubblesort/sort.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func split(s string) []string {
	if s == "" {
		return []string{}
	} else {
		return strings.Split(s, " ")
	}
}

func arrItoa(a []int) []string {
	res := []string{}
	for _, v := range a {
		val := strconv.Itoa(v)
		res = append(res, val)
	}
	return res
}

func sort(a []int) []int {
	for i := len(a) - 1; i >= 0; i-- {
		swapped := false
		for j := 0; j < i; j++ {
			if a[j] > a[j+1] {
				swapped = true
				a[j], a[j+1] = a[j+1], a[j]
			}
		}
		if !swapped {
			return a
		}
	}
	return a
}

func arrAtoi(a []string) []int {
	res := []int{}
	for _, v := range a {
		val, _ := strconv.Atoi(v)
		res = append(res, val)
	}
	return res
}

func main() {
	scan := bufio.NewScanner(os.Stdin)
	scan.Scan()
	list := arrAtoi(split(scan.Text()))
	fmt.Println(strings.Join(arrItoa(sort(list)), " "))
}
